File: PythonPhaseGateOne/checkout_app.py
from datetime import datetime

def get_product_details(item_name, item_quantity, item_price, item_total):
	product_details = {}
	product_details['ITEM'] = item_name;
	product_details['QTY'] = item_quantity;
	product_details['PRICE'] = item_price;
	product_details['TOTAL'] = item_total;
	return product_details


def get_sub_total(user_cart):
	sum = 0;
	for number in range(0, len(user_cart)):
		for detail in user_cart[number]:
			if detail == "TOTAL":
				sum += user_cart[number][detail]
	return sum

	
def get_discount_of(sub_total, discount):
	discount_amount = sub_total * (discount / 100)
	return discount_amount


def get_vat_of(sub_total):
	vat_amount = sub_total * (17.50 / 100)
	return vat_amount



def get_bill_total(sub_total, discount_amount, vat_amount):
	return (sub_total - discount_amount) + vat_amount


def get_balance(amount_received, bill_total):
	return amount_received -  bill_total





user_cart = []



user_name = input("What is the customer's Name? ")

item_name = input("What did the customer buy? ")
item_quantity = int(input("How many pieces? "))
item_price = int(input("How much per unit? "))
item_total = item_price * item_quantity


user_cart.append(get_product_details(item_name, item_quantity, item_price, item_total))
add_more_items = input("Add more Items? ")
while add_more_items.lower() == "yes":
	item_name = input("What did the customer buy? ")
	item_quantity = int(input("How many pieces? "))
	item_price = int(input("How much per unit? "))
	item_total = item_price * item_quantity
	user_cart.append(get_product_details(item_name, item_quantity, item_price, item_total))
	add_more_items = input("Add more Items? ")


cashier_name = input("What is your name? ")
discount = int(input("How much discount will he get? "))
sub_total = get_sub_total(user_cart)
discount_amount = get_discount_of(sub_total, discount)
vat_amount = get_vat_of(sub_total)
bill_total = get_bill_total(sub_total, discount_amount, vat_amount)

name_of_store = """SEMICOLON STORES
				MAIN BRANCH
			"""
store_address = "312, HERBERT MACAULAY WAY, SABO YABA, LAGOS"
contact = "03293828343"
current_datetime = datetime.now()


print(name_of_store)
print("LOCATION: " + store_address)
print("TEL: " + contact)
print(f'Date {current_datetime}')
print("Cashier: " + cashier_name)

print("Customer Name: " + user_name)


 
print("--" * 10)
columns = user_cart[0].keys()
print("\t" * 5 + "\t".join(columns))
for row in user_cart:
	print("\t".join(str(value) for value in row.values()))

print("\t" * 4 + "Sub Total", sub_total)
print("\t" * 4 + "Discount", discount_amount)
print("\t" * 4 + "VAT @ 17.50%", vat_amount)
print("\t" * 4 + "Bill Total", bill_total)

print("THIS IS NOT A RECEIPT KINDLY PAY", bill_total)




amount_received = int(input("How much did the customer give to you? "))
balance = get_balance(amount_received, bill_total)

print("--" * 10)
columns = user_cart[0].keys()
print("\t" * 5 + "\t".join(columns))
for row in user_cart:
	print("\t".join(str(value) for value in row.values()))

print("\t" * 4 + "Sub Total: ", sub_total)
print("\t" * 4 + "Discount: ", discount_amount)
print("\t" * 4 + "VAT @ 17.50%: ", vat_amount)
print("\t" * 4 + "Bill Total: ", bill_total)
print("\t" * 4 + "Amount Paid: ", amount_received)
print("\t" * 4 + "Balance: ", balance)

print("THANK YOU FOR YOUR PATRONAGE")

File: JavaScriptPhaseGateOne/CheckOutApp.js
function getProductDetails(itemName, itemQuantity, itemPrice, itemTotal){
	let productDetails = {};
	productDetails.ITEM = itemName;
	productDetails.QTY = itemQuantity;
	productDetails.PRICE = itemPrice;
	productDetails.TOTAL = itemTotal;
	return productDetails;
}

function getSubTotal(userCart){
	let sum = 0;
	for(let index = 0; index < userCart.length; index++){
		for(let detail in userCart[index]){
			if (detail == "TOTAL"){
				sum += userCart[index][detail];
			}
		}
	}
	return sum;
}
	
function getDiscountOf(subTotal, discount){
	let discountAmount = subTotal * (discount / 100);
	return discountAmount;
}

function getVATOf(subTotal){
	let VATAmount = subTotal * (17.50 / 100);
		return VATAmount;
}


function getBillTotal(subTotal, discountAmount, VATAmount){
	return (subTotal - discountAmount) + VATAmount;
}

function getBalance(amountReceived, billTotal){
	return amountReceived - billTotal;
}



const prompt = require('prompt-sync')();
//let productDetails = {};
let userCart = [];




let userName;
let itemName;
let itemQuantity;
let itemPrice;
let itemTotal;
let discount;
let cashierName;


userName = prompt("What is the customer's Name? ");

itemName = prompt("What did the customer buy? ");
itemQuantity = prompt("How many pieces? ");
itemPrice = prompt("How much per unit? ");
itemTotal = itemPrice * itemQuantity;


userCart.push(getProductDetails(itemName, itemQuantity, itemPrice, itemTotal));
let addMoreItems = prompt("Add more Items? ");
while(addMoreItems.toLowerCase() == "yes"){
	itemName = prompt("What did the customer buy? ");
	itemQuantity = prompt("How many pieces? ");
	itemPrice = prompt("How much per unit? ");
	itemTotal = itemPrice * itemQuantity;
	userCart.push(getProductDetails(itemName, itemQuantity, itemPrice, itemTotal));
	addMoreItems = prompt("Add more Items? ");
}

cashierName = prompt("What is your name? ");
discount = prompt("How much discount will he get? ");
let subTotal = getSubTotal(userCart);
let discountAmount = getDiscountOf(subTotal, discount);
let VATAmount = getVATOf(subTotal);
let billTotal = getBillTotal(subTotal, discountAmount, VATAmount);

let nameOfStore = `SEMICOLON STORES
				MAIN BRANCH
			`;
let storeAddress = "312, HERBERT MACAULAY WAY, SABO YABA, LAGOS"
let contact = "03293828343";
let date = new Date();


console.log(nameOfStore);
console.log("LOCATION: " + storeAddress);
console.log("TEL: " + contact);
console.log("Date: " + date);
console.log("Cashier: " + cashierName);

console.log("Customer Name: " + userName);




console.log("--".repeat(10));
let columns = Object.keys(userCart[0]);
console.log("\t".repeat(5) + columns.join("\t"));
userCart.forEach(row => {console.log("\t".repeat(4) + Object.values(row).join("\t"));
});
console.log("\t".repeat(4) + "Sub Total" + subTotal);
console.log("\t".repeat(4) + "Discount" + discountAmount);
console.log("\t".repeat(4) + "VAT @ 17.50%" + VATAmount);
console.log("\t".repeat(4) + "Bill Total" + billTotal);

console.log("THIS IS NOT A RECEIPT KINDLY PAY" + billTotal);





let amountReceived = prompt("How much did the customer give to you? ");
let balance = getBalance(amountReceived, billTotal)

console.log("--".repeat(10));
columns = Object.keys(userCart[0]);
console.log("\t".repeat(5) + columns.join("\t"));
userCart.forEach(row => {console.log("\t".repeat(4) + Object.values(row).join("\t"));
});
console.log("\t".repeat(4) + "Sub Total: " + subTotal);
console.log("\t".repeat(4) + "Discount: " + discountAmount);
console.log("\t".repeat(4) + "VAT @ 17.50%: " + VATAmount);
console.log("\t".repeat(4) + "Bill Total: " + billTotal);
console.log("\t".repeat(4) + "Amount Paid: " + amountReceived);
console.log("\t".repeat(4) + "Balance: " + balance);


console.log("THANK YOU FOR YOUR PATRONAGE");















































